src/main/java/eu/xenit/gradle/tasks/DockerfileWithCopyTask.java

```diff
@@ -0,0 +1,57 @@
+package eu.xenit.gradle.tasks;
+
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
+import java.nio.file.Path;
+import org.gradle.api.file.CopySpec;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.TaskAction;
+
+public class DockerfileWithCopyTask extends Dockerfile {
+    private int copyFileCounter = 0;
+
+    private final CopySpec copyFileCopySpec;
+
+    private String createCopyFileStagingDirectory() {
+        copyFileCounter++;
+        return "copyFile/"+copyFileCounter;
+    }
+
+    public void copyFile(java.io.File file, String destinationInImage) {
```

Comment:
Can we find another name? Because now, you would think it does more or less the same thing than the parent copyFile(String, String) method.

Reply:
Renamed to `smartCopy`